### ISSUE TITLE
Issue - Table Headers are not read using screen reader - Tab Key doesn't work on table headers. 

### ISSUE THREAD

**Issue by santoshmact** [ant-design/ant-design#24772]
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[https://6n45e.csb.app/](https://6n45e.csb.app/)

### Steps to reproduce
1. Navigate to sandbox url.
2. Use Tab key to navigate to the Table Headers. Unable to navigate using Tab Key.
3. Use JAWS or any other screen reader, header name is not announced.

### What is expected?
Should be able to navigate to headers using Tab Key.
Table Header should be announced using screen readers.

### What is actually happening?
Unable to navigate to table headers using Tab Key.
Table Headers are not announced using screen readers.

| Environment | Info |
|---|---|
| antd | 4.3.1 |
| React | 16.12.0 |
| System | window |
| Browser | Goggle Chrome 83.0 |

**Comment by santoshmact** [comment]
Team - The issue is stopping us to use ant table control as the headers are not accessible. 
Please look into the issue or let us know the alternative.

**Comment by santoshmact** [comment]
@yoyo837 - Any update on this issue. Please look into the issue or let us know the alternative.

**Comment by jimmieego** [comment]
The header is a `div`. It needs to have `tabindex="0"` to be in the tab order. It probably also needs to respond to <kbd>Enter</kbd> key.

**Comment by santoshmact** [comment]
Yes. I agree. 

It should respond to Enter key so the sorting functionality can be triggered on Enter key press.

**Comment by dgreene1** [comment]
Is tab the correct mechanism to navigate tables? My expectation after [following the NVDA training for tables](https://webaim.org/articles/nvda/tables.htm) is that you would use `t` to find the table and `shift
+t` to get the previous table.

Sadly, that's not working either with antd. So while the expected results of this ticket might be wrong, the essence of the ticket is correct: antd's tables are not accessible.
